tsuikm/usdc-dashboard
# Feature/blacklist unblacklist functionality

# Contributors
@constanceshi @tetsuroescudero 

# Relevant issue
#96 

# Summary of change
- add ethereum requests to actually blacklist/unblacklist addresses when the corresponding button is pressed
- subscribe once to a blacklist/unblacklist event to update the UI to reflect the new state once the transaction to blacklist/unblacklist is processed successfully

# Testing plan
- updated web3 mock and BlacklisterControl.spec.js to test for internal state changes
- manually tested functionality

# Testing instructions
-  ask tetsu to make you the blacklister (since there can only be one at a time) for our test token
- export USDC contract address in utils/constants.js as the Test token address (0xfc7e3a2554e2d4b23e41c81b14065ee31009cc31)
- navigate to the roles/blacklister route
- input a random address and check the status
- click blacklist/unblacklist to fire events, which will cause metamask to prompt you to connect your account and with a transaction
- you can decrease the gas value so it takes less eth, but it will be slower the lower it is
- you can also go to your pending transactions and click speed up and pay a bit more eth after you lower the gas price
- once you are notified the transaction has gone through, the status of the address being blacklisted/unblacklisted should change automatically on the screen, and rechecking the status should also reflect the new state
- from here you can toggle the blacklist/unblacklist button to redo/undo your action and process a new transaction to blacklist/unblacklist, following the instructions above
